packages/core/src/odata/common/filter/filter-function.ts

```diff
@@ -49,14 +49,47 @@ export function filterFunction<EntityT extends EntityBase>(
   return new NumberFilterFunction(
     functionName,
     parameters,
-    returnTypeToEdmType[returnType]
+    numberReturnTypeMapping[returnType]
   );
 }
 
-const returnTypeToEdmType: MapType<
+// eslint-disable-next-line valid-jsdoc
+/**
+ * @hidden
+ */
+export function createFilterFunction<EntityT extends EntityBase>(
```

Comment:
This looks like code duplication with respect to `filterFunction()` a few lines above.

![image](https://user-images.githubusercontent.com/56544999/86100420-1444f100-bab9-11ea-8ba3-6d069a045cc3.png)

Reply:
Oops, I meant to call it in the function above

Reply:
Fixed, anyways unfortunately you will find something very similar in v4. I did not find a way to share the overloads between two implementations (https://stackoverflow.com/questions/62642368/how-to-share-typescript-overloads-across-modules).

Reply:
I think I mentioned this already on the initial PR #217 that I find the name `filterFunction` not optimal since ti is creating a filter function. Could we not use the `createFilterFunction` but I guess this would be a breaking change now.

Reply:
Yes it would be breaking. This has been there since before we open sourced.
Anyways, we could still deprecate this, but I prefer the current name. This is meant to be used by our users directly:
```ts
.filter(filterFunction('fn', 'int').equals(3))
```
In this context I like it more than:
```ts
.filter(createFilterFunction('fn', 'int').equals(3))
```